### PR TITLE
Fix launching the navigator

### PR DESCRIPTION
**Pull request by mickael-menu**
Improves the way navigator activities are launched, and fix the initial locator.

This relates to the PR implementing navigator fragments: https://github.com/readium/r2-navigator-kotlin/pull/148

### CODE REVIEW

File: r2-testapp/src/main/java/org/readium/r2/testapp/library/LibraryActivity.kt [hunk 1]
```diff
@@ -592,7 +594,14 @@ abstract class LibraryActivity : AppCompatActivity(), BooksAdapter.RecyclerViewC
                     prepareToServe(it, file)
                     progress.dismiss()
                     navigatorLauncher.launch(
-                        NavigatorContract.PublicationData(file, it, deleteOnResult = remoteUrl != null )
+                        NavigatorContract.Input(
+                            file = file,
+                            publication = it,
+                            bookId = book.id,
+                            initialLocator = book.id?.let { id -> booksDB.books.currentLocator(id) },
+                            drm = it.isProtected,
```
Comment (qnga) [inline r2-testapp/src/main/java/org/readium/r2/testapp/library/LibraryActivity.kt:602]: What's the purpose of passing `drm` in the intent? `publication.isProtected` would be available from the navigator.

File: r2-testapp/src/main/java/org/readium/r2/testapp/epub/EpubActivity.kt [hunk 2]
```diff
@@ -233,11 +233,6 @@ class EpubActivity : R2EpubActivity(), CoroutineScope, NavigatorDelegate/*, Visu
         })
         search_listView.adapter = searchResultAdapter
         search_listView.layoutManager = LinearLayoutManager(this)
-
-        // Loads the last read location
-        booksDB.books.currentLocator(bookId)?.let {
-            go(it, false, {})
-        }
     }
```
Comment (qnga) [inline r2-testapp/src/main/java/org/readium/r2/testapp/epub/EpubActivity.kt:236]: Was this some useless legacy code?

Comment (mickael-menu) [reply r2-testapp/src/main/java/org/readium/r2/testapp/epub/EpubActivity.kt:236]: No, now it works by providing a `initialLocator` with the `NavigatorContract`. Positioning the initial location is now handled on the navigator side in this PR https://github.com/readium/r2-navigator-kotlin/pull/148. I think it's still missing for the other formats.

File: r2-testapp/src/main/java/org/readium/r2/testapp/library/LibraryActivity.kt (hunk 1 above)
Comment (mickael-menu) [reply r2-testapp/src/main/java/org/readium/r2/testapp/library/LibraryActivity.kt:602]: Yes you're right, I'll modify the legacy activities instead.

**Pull request merged**